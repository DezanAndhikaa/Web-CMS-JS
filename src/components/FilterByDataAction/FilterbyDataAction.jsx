

import React from 'react';
import './FilterbyDataAction.scss';
import { Badge } from '@material-ui/core';
import { IconApprove, IconHistory, IconInputLT } from '../../assets/icons';
import { MenuList, MenuItem, Paper, Typography, withStyles } from '@material-ui/core';
import { Menu } from '../../constants';
import roleService from "../../utils/roleService.helper";

const RoleUser = new roleService();
const DotBadges = withStyles(theme => ({
	badge: {
    top: -1,
    border: '2px solid white',
    width: 'auto',
    minWidth: '15px',
    height: '15px',
    borderRadius: '50%'
	},
}))(Badge);

class FilterbyDataAction extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      displayMenu: false,
    };
  }

  showDropdownMenu = (event) => {
    event.preventDefault();
    this.setState({ displayMenu: true }, () => {
      document.addEventListener('click', this.hideDropdownMenu);
    });
  }

  hideDropdownMenu = async() => {
    this.setState({ displayMenu: false }, () => {
      document.removeEventListener('click', this.hideDropdownMenu);
    });
  }

  selectItem = (item) => {
    this.props.onSelectAction(this.props.onSelectActionType, item);
  }

  handleClick = (menu, tab) => {
    this.props.push({
      pathname: menu,
      whichTab: tab
    });
    window.localStorage.setItem('whichTab', tab);
  }

  renderList(){
    return(
      <Paper className={this.props.idStatus === "DetailSite" ? "list-status-site" : "list-status" }>
        {Number(RoleUser.role()) === 1 ? 
          <MenuList>
            <MenuItem>
              <Typography 
                className="list-item-status"
                variant="inherit"
                onClick={ () =>{ localStorage.getItem('subMenu') === "/webcms/planning/ho" ? this.handleClick(Menu.PLANNING_HO_STATUS, 'service') 
                : this.handleClick(Menu.PLANNING_SITE_STATUS, 'service') }}
              >
                Service Order
              </Typography>
            </MenuItem>
          </MenuList>
        : <MenuList>
            <MenuItem>
              <Typography className="list-item-status" variant="inherit" onClick={ () => this.handleClick(Menu.PLANNING_SITE_STATUS, 'service')}>Service Order</Typography>
            </MenuItem>
          </MenuList>
        }
      </Paper>
    )
  }

  render() {
    if(this.props.titles === "Status"){
      return (
        <div className="dropdown">
          <div className="dropdown-button" onClick={this.showDropdownMenu}>
          <div className="dropdown-selected-item">
            <div className="tracking-history">
              <DotBadges color="secondary" badgeContent="" anchorOrigin={{ vertical: 'top', horizontal: 'left', }}>
                  <img src={IconApprove} className="icon-history" alt="" /><span className="label-status">Status</span>
              </DotBadges>
            </div>
          </div>
        </div>
          { 
            this.state.displayMenu && this.renderList()  
          }
        </div>
      );
    }
    else if(this.props.titles === "Tracking History"){
      return (
        <div className="dropdown">
          <div className="dropdown-button" onClick={() => this.handleClick(Menu.PLANNING_TRACKING_HISTORY, 'tracking')}>
            <div className="dropdown-selected-item">
              <div className="tracking-history">
                <img src={IconHistory} className="icon-history" alt="" /><span className="label-history">Tracking History</span>
              </div>
            </div>
          </div>
        </div>
      );
    }
    else if(this.props.titles === "Input Lifetime"){
      return (
        <div className="dropdown">
          <div className="dropdown-button">
            <div className="dropdown-selected-item">
              <div className="input-lt" onClick={ () => this.handleClick(Menu.PLANNING_INPUT_LIFETIME, 'lifetime') }>
                <img src={IconInputLT} className="icon-lt" alt="" /><span className="label-lt">Input Lifetime</span>
              </div>
            </div>
          </div>
        </div>
      );
    }
  }
}

export default FilterbyDataAction;