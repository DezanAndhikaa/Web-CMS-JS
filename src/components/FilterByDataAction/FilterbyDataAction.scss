@import '../../themes/variables';
@import '../../themes/functions';

.dropdown {
    // position: absolute;
    display: flex;
    // background-color: #ffffff;
    // width: vw(233.22);
    // height: vw(45.07);
    // cursor: pointer;
    // margin-left: 19.2vw;
    // border-radius:vw(15) !important;
    // box-shadow: 0px 0.2vw 0.6vw rgba(0, 0, 0, 0.161);
    .dropdown-button {
        border-radius: vw(10);
        display: flex;
        // align-items: center;
        // padding-left: 10%;
        height: vw(45.07);
        .dropdown-selected-item{            
            margin: 0 0 0 vw(19);
            display: flex;
            flex-direction: row;
            white-space: nowrap;
            align-items: center;
            font-family: 'Lato-Medium';
            font-size: vw(16);
            line-height: vw(17);
            font-weight: 500;
            // overflow: hidden;
            color: black;
            cursor: pointer;
            .expand-icon-container{
                background-image: url('../../assets/icons/folder.png');
                background-color: map-get($colors, third);
                background-size: 50%;
                background-repeat: no-repeat;
                background-position: center;
                text-align: center;
                position: absolute;
                right: 0;
                height: 100%;
                width: vw(45.07);
                display: flex;
                border-bottom-right-radius: vw(10) !important;
                border-top-right-radius: vw(10) !important;
                .expand-icon{
                    font-size: vw(25);
                }
            }
            .tracking-history{
                display: flex;
                flex-direction: row;
                align-self: flex-end;
                // cursor: pointer;
                background-color: transparent;
                align-items: center;
                .icon-history{
                    margin-right: 10px; //vw(10)
                }
                .label-history{
                    font-family: 'Lato-Bold';
                    font-size: 14px; //vw(20)
                    line-height: 18px; // vw(24)
                    color: #3C3B3B;
                    text-transform: capitalize;
                }
                .label-history:hover{
                color: #FC9517;
                }
            }
            .input-lt{
                display: flex;
                flex-direction: row;
                align-self: flex-end;
                // cursor: pointer;
                background-color: transparent;
                align-items: center;
                .icon-lt{
                    // height: 18px; //vw(25) !important
                    margin-right: 10px; //vw(10)
                }
                // .icon-history:hover{
                //     background-image: url('../../assets/icons/clock.png');
                // }
                .label-lt{
                    font-family: 'Lato-Bold';
                    font-size: 14px; //vw(20)
                    line-height: 18px; // vw(24)
                    color: #3C3B3B;
                    text-transform: capitalize;
                }
                .label-lt:hover{
                color: #FC9517;
                }
            }
        }
    }
    .dropdown-button:focus{
        border-style: solid;
        border-color: map-get($colors, third);
    }
}

.list-items-fbdt {
    top: vw(50);
    background-color: white;
    position: absolute;
    box-shadow: 0px 0.3vw 0.6vw 0px rgba(0,0,0,0.2);
    z-index: 1;
    display: flex;
    justify-content: center;
    align-items: center;
    align-content: center;
    align-self: center;
    .list-item-fbdt {
        // padding: vw(6) vw(19);
        color: #3C3B3B;
        font-family: 'Lato-Bold';
        font-size: vw(20);
        // line-height: vw(17);
    }
    // .list-item-fbdt:hover {
    //     background-color: #e5e5e5;
    //     color: #3C3B3B;
    // }
}

.list-items-fbdt::before {
    content:"";
    position: absolute;
    right: 45%;
    top: -5px;
    width: 0;
    height: 0;
    border-style: solid;
    border-width: 0 8px 8px 8px;
    border-color: transparent transparent white transparent;
    z-index:9998;
}

// .list-items {
//     margin: 0;
//     right: 0px;
//     width: vw(233.22);
//     height: vw(150);
//     border-radius:vw(10) !important;
//     background-color: white;
//     position: absolute;
//     z-index: 100;
//     box-shadow: 0px 0.3vw 0.6vw 0px rgba(0,0,0,0.2);
//     color: rgb(0, 0, 0);
//     display: flex;
//     align-items: center;
//     justify-content: center;
//     flex-wrap: wrap;
//     margin-top: 150px;
//     .button{
//         width: vw(173.22);
//         height: vw(45.07);
//         border-radius: vw(10) !important;
//         font-size: 10px;
//         text-align: center;
//         text-transform: capitalize;
//         margin-top: 0.5vw;
//         padding: inherit;
//     }
//     .button-plan-del{
//         width: vw(173.22);
//         height: vw(45.07);
//         border-radius: vw(10) !important;
//         font-size: 10px;
//         text-align: center;
//         text-transform: capitalize;
//         margin-bottom: 0.5vw;
//         padding: inherit;
//     }
//     .button:focus{
//         color: map-get($colors, primary);
//         border-style: solid;
//         border-color: map-get($colors, primary);
//     }
// }

@media (min-width: 600px) and (max-width: 1025px) {
    .dropdown {
        position: relative;
        display: inline-block;
        background-color: #ffffff;
        width: vw-tab(220);
        height: vw-tab(33);
        border-radius: vw-tab(5);
        box-shadow: 0px 0.2vw 0.6vw rgba(0, 0, 0, 0.161);
        .dropdown-button {
            position: relative;
            border-radius: vw-tab(5);
            display: flex;
            align-items: center;
            height: vw-tab(33);
            .dropdown-selected-item{
                margin: 0 0 0 vw-tab(19);
                flex: 1;
                position: relative;
                height: vw-tab(33);
                display: flex;
                flex-direction: row;
                white-space: nowrap;
                align-items: center;
                font-family: 'Lato-Medium';
                font-size: vw-tab(14);
                line-height: vw-tab(17);
                font-weight: 500;
                overflow: hidden;
                color: black;
                .expand-icon-container{
                    background-color: map-get($colors, primary);
                    position: absolute;
                    right: 0;
                    top: 0;
                    height: vw-tab(33);
                    width: 20%;
                    display: flex;
                    align-items: center;
                    justify-content: center;
                    border-bottom-right-radius: vw-tab(5);
                    border-top-right-radius: vw-tab(5);
                    .expand-icon{
                        font-size: vw-tab(25);
                    }
                }
            }
        }    
        .list-items {
            list-style-type: none;
            margin: 0;
            padding: vw-tab(10) 0 0 0;
            top: vw-tab(35);
            right: 0px;
            width: vw-tab(220);
            height: vw-tab(170);
            overflow-x: hidden;
            overflow-y: scroll;
            background-color: white;
            position: absolute;
            box-shadow: 0px 0.3vw 0.6vw 0px rgba(0,0,0,0.2);
            z-index: 1;
            font-weight: 500;
            color: #3C3B3B;
            .list-item {
                padding: vw-tab(12) vw-tab(19);
                font-family: 'Lato-Medium';
                font-size: vw-tab(14);
                line-height: vw-tab(17);
                text-overflow: ellipsis;
                color: black;
                font-size: vw-tab(14);
            }
            .list-item:hover {
                background-color: #e5e5e5;
                color: #000;
            }
        }
    }
}


@media (max-width: 599px) {
    .list-items {
        list-style-type: none;
        display: flex;
        flex-direction: column;
        margin-bottom: vw-mobile(2);
        top: vw-mobile(2);
        padding: vw-mobile(10) 0 0 0;
        width: 100%;
        height: vw-mobile(170);
        overflow-x: hidden;
        overflow-y: scroll;
        background-color: white;
        border-radius: vw-mobile(5);
        position: relative;
        box-shadow: 0px vw-mobile(3) vw-mobile(6) rgba(0,0,0,0.2);
        font-weight: 500;
        color: #3C3B3B;
        .list-item {
            padding: vw-mobile(12) vw-mobile(19);
            font-family: 'Lato-Medium';
            font-size: vw-mobile(14);
            line-height: vw-mobile(17);
            text-overflow: ellipsis;
            color: black;
            font-size: vw-mobile(14);
        }
        .list-item:hover {
            background-color: #e5e5e5;
            color: #000;
        }
    }
    .dropdown {
        position: relative;
        display: inline-block;
        background-color: #ffffff;
        width: 100%;
        height: vw-mobile(33);
        border-radius: vw-mobile(5);
        box-shadow: 0px vw-mobile(3) vw-mobile(6) rgba(0,0,0,0.2);
        .dropdown-button {
            position: relative;
            border-radius: vw-mobile(5);
            display: flex;
            align-items: center;
            height: vw-mobile(33);
            .dropdown-selected-item{
                margin: 0 0 0 vw-mobile(19);
                height: vw-mobile(33);
                position: relative;
                flex: 1;
                display: flex;
                flex-direction: row;
                white-space: nowrap;
                align-items: center;
                font-family: 'Lato-Medium';
                font-size: vw-mobile(14);
                line-height: vw-mobile(17);
                font-weight: 500;
                overflow: hidden;
                color: black;
                .expand-icon-container{
                    background-color: map-get($colors, primary);
                    position: absolute;
                    right: 0;
                    top: 0;
                    height: vw-mobile(33);
                    width: 20%;
                    display: flex;
                    align-items: center;
                    justify-content: center;
                    border-bottom-right-radius: vw-mobile(5);
                    border-top-right-radius: vw-mobile(5);
                    .expand-icon{
                        font-size: vw-mobile(25);
                    }
                }
            }
        }            
    } 
}