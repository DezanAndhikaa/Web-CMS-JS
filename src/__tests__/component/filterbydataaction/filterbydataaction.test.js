import React from 'react';
import Enzyme, { shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import FilterbyDataAction from '../../../components/FilterbyDataAction/FilterbyDataAction';

Enzyme.configure({adapter:new Adapter()});
it("render filter by data action correctly", () => {
    const wrapper = shallow (
        <FilterbyDataAction />
    )
    expect(wrapper).toMatchSnapshot();
})