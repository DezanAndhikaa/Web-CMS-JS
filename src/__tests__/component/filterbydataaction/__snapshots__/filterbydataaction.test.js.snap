// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render filter by data action correctly 1`] = `ShallowWrapper {}`;
